### ISSUE TITLE
Reloaded slices and projections cannot be visualized with SlicePlot and ProjectionPlot as advertised in the docs

### ISSUE THREAD

**Issue by ngoldbaum** [yt-project/yt#1444]
According to this page in the docs:

http://yt-project.org/doc/analyzing/saving_data.html#spatial-plots

Something like the following script should work:

```python
import yt
ds = yt.load('IsolatedGalaxy/galaxy0030/galaxy0030')
plot = yt.SlicePlot(ds, 'z', 'density')
plot.data_source.save_as_dataset('slice.h5')
ds_slice = yt.load('slice.h5')
p = yt.SlicePlot(ds_slice, 'z', 'density')
```

Unfortunately, right now it errors out with the following traceback:

```
---------------------------------------------------------------------------
YTInvalidFieldType                        Traceback (most recent call last)
/Users/goldbaum/Documents/yt-git-fixes/yt/mods.py in <module>()
----> 1 p = yt.SlicePlot(ds_slice, 'z', 'density')

/Users/goldbaum/Documents/yt-git-fixes/yt/visualization/plot_window.py in SlicePlot(ds, normal, fields, axis, *args, **kwargs)
   2014             del kwargs['north_vector']
   2015
-> 2016         return AxisAlignedSlicePlot(ds, normal, fields, *args, **kwargs)

/Users/goldbaum/Documents/yt-git-fixes/yt/visualization/plot_window.py in __init__(self, ds, axis, fields, center, width, axes_unit, origin, right_handed, fontsize, field_parameters, window_size, aspect, data_source)
   1328                            center=center, data_source=data_source)
   1329             slc.get_data(fields)
-> 1330         validate_mesh_fields(slc, fields)
   1331         PWViewerMPL.__init__(self, slc, bounds, origin=origin,
   1332                              fontsize=fontsize, fields=fields,

/Users/goldbaum/Documents/yt-git-fixes/yt/visualization/plot_window.py in validate_mesh_fields(data_source, fields)
    138
    139     if len(invalid_fields) > 0:
--> 140         raise YTInvalidFieldType(invalid_fields)
    141
    142

YTInvalidFieldType:
SlicePlot, ProjectionPlot, and OffAxisProjectionPlot can only plot fields that
are defined on a mesh, but received the following particle fields:

    [('all', 'density')]

Did you mean to use ParticlePlot or plot a deposited particle field instead?
```

So either the documentation needs to be updated to indicate that you need to use ParticlePlot if you do this or this used to work and something has broken in the meantime. @brittonsmith do you remember whether this ever worked?

**Comment by brittonsmith** [comment]
I can confirm that this did work when I put it in the documentation, but I guess it wasn't being tested.  I won't have time to look at this for a couple weeks, but I can try then.

**Comment by ngoldbaum** [comment]
Verified just now that it works on the changeset that added the feature. I'm going to see if I can bisect now...

**Comment by ngoldbaum** [comment]
Bisects to this commit:

```
commit 8b7bd89c55ef6c66bfe54ff4494da771f82d532a
Author: Nathan Goldbaum <ngoldbau@illinois.edu>
Date:   Sat Jul 16 10:36:56 2016 -0500

    Print a more informative error message when PlotWindow is passed a particle field

    Closes #878

    --HG--
    branch : yt
```
